codec.go

```diff
@@ -49,6 +51,30 @@ type Codec interface {
 	Unmarshal([]byte, any) error
 }
 
+// StableCodec is an extension to Codec for serializing with stable output.
+type StableCodec interface {
```

Comment:
Is adding a new interface here the right call? This feels like we're polluting the namespace a bit.

Reply:
Not sure. Me and Akshay discussed potential approaches. "Interface smuggling" seemed to be a sensible way to add a new optional extension like this.

Reply:
As a starting position, I'd love to keep this interface unexported and document the optional methods on the exported `Codec` interface. Can we see how that works out, and whether it ends up feeling too implicit?

In practice, I expect that _very_ few users will end up both customizing codecs and caring about GETs.

Reply:
Can we unexport it?

Reply:
Done.